tests/Mapper/ORM/DoctrineORMMapperTest.php

```diff
@@ -14,17 +14,41 @@
 namespace Sonata\Doctrine\Tests\Mapper\ORM;
 
 use Doctrine\ORM\Mapping\ClassMetadataInfo;
-use PHPUnit\Framework\TestCase;
 use Sonata\Doctrine\Mapper\Builder\ColumnDefinitionBuilder;
+use Sonata\Doctrine\Mapper\Builder\OptionsBuilder;
+use Sonata\Doctrine\Mapper\DoctrineCollector;
 use Sonata\Doctrine\Mapper\ORM\DoctrineORMMapper;
+use Sonata\Doctrine\Tests\Fixtures\App\Entity\TestEntity;
+use Sonata\Doctrine\Tests\Fixtures\App\Entity\TestRelatedEntity;
+use Symfony\Bundle\FrameworkBundle\Test\KernelTestCase;
 
-class DoctrineORMMapperTest extends TestCase
+class DoctrineORMMapperTest extends KernelTestCase
 {
     /**
      * @var ClassMetadataInfo
      */
     private $metadata;
 
+    public static function setUpBeforeClass(): void
+    {
+        $options = OptionsBuilder::createManyToOne('relation', TestRelatedEntity::class)
+            ->add('joinColumns', [['referencedColumnName' => 'id']]);
+
+        $override = OptionsBuilder::create()
```

Comment:
Side note: we deprecrated the direct create here: https://github.com/sonata-project/sonata-doctrine-extensions/pull/177
But for adding overrides, you need to use the options builder and it might be an override of a non relation field. So the methods currently available are not suitable for this use case. Maybe we need to add one more method or undeprecate this create.

Reply:
cc @franmomu

Reply:
Maybe we could create a class for override, I've never used them, which are the options? And maybe since there has been no release we could create AssociationBuilder which extends OptionsBuilder (to be BC) and has all the methods I added.

Reply:
IMO that should belong to another PR. Is this one ready!?

Reply:
Sure it should belong to another PR, just asking about docs to create that class.